pkg/utils/tests/tests.go

```diff
@@ -68,3 +72,34 @@ func RequireSignal(t *testing.T, ch <-chan struct{}, failMsg string) {
 		t.Fatal(failMsg)
 	}
 }
+
+// Convenience method to setup a client connection with a default timeout of 5 seconds
+func SetupClientConnWithOptsAndTimeout(t *testing.T, target string, opts ...grpc.DialOption) (*grpc.ClientConn, error) {
```

Comment:
If you like this convenience method, I can refactor the other tests to consume it.